web/modules/custom/joinup_group/src/Entity/GroupTrait.php

```diff
@@ -12,40 +12,15 @@
  */
 trait GroupTrait {
 
-  /**
-   * Static cache of memberships for this group, keyed by user ID.
-   *
-   * @var \Drupal\og\OgMembershipInterface[]|NULL[]
-   */
-  protected $memberships;
-
   /**
    * {@inheritdoc}
    */
   public function getMembership(int $uid): ?OgMembershipInterface {
     assert(is_subclass_of($this, ContentEntityBase::class), __TRAIT__ . ' is intended to be used in bundle classes for content entities.');
-    if (!isset($this->memberships[$uid])) {
-      $og_membership_storage = $this->entityTypeManager()->getStorage('og_membership');
-
-      $query = $og_membership_storage
-        ->getQuery()
-        ->condition('uid', $uid)
-        ->condition('entity_type', $this->getEntityTypeId())
-        ->condition('entity_id', $this->id());
-
-      $membership_ids = $query->execute();
-      if (empty($membership_ids)) {
-        return NULL;
-      }
-      $membership_id = reset($membership_ids);
-
-      /** @var \Drupal\og\OgMembershipInterface|NULL $membership */
-      $membership = $og_membership_storage->load($membership_id);
-
-      $this->memberships[$uid] = $membership;
-    }
 
-    return $this->memberships[$uid];
+    /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+    $membership_manager = \Drupal::service('og.membership_manager');
+    return $membership_manager->getMembership($this, $uid, [OgMembershipInterface::STATE_ACTIVE]);
```

Comment:
With the previous code all memberships, regardless of their state, were returned. Now, I see we're limiting to the active states. Shouldn't we better use `OgMembershipInterface::ALL_STATES`?

Reply:
Yes you are right, we should also return blocked memberships, otherwise calling code cannot retrieve a blocked membership if they need it.

Reply:
I updated this, but I made it an option that can be passed by the calling code, and defaulting to not returning a blocked membership. I remember discussing this flag with @amitaibu at an earlier occasion and he convinced me that it is safer to exclude blocked users by default. The reasoning is as follows:

If blocked users are excluded by default, then if a programming mistake is made (e.g. the developer is not considering that a blocked membership might be returned) then blocked users are not going to be given access to some new feature by default. This is the most common case, blocked users are treated the same way as non-members for most functionalities. So the worst case scenario is that a blocked user is not given access to something they should have access to. This can be annoying but is not critical.

On the other hand, if we would default to returning a blocked membership, then if a programming mistake is made, then a blocked user would be accidentally given access to something they should __not__ have access to. This can lead to critical problems like information disclosure.

So it would be safer like this, and it is also consistent with `MembershipManagerInterface::getMembership()`.

Is this OK for you?

Reply:
@pfrenssen yes, makes totally sense

Reply:
I don't remember that specific conversation, but yeah - it makes a lot of sense :smile_cat: